### PR TITLE
fix: State propagation during module CR deletion

### PR DESCRIPTION
**Pull request by ruanxin**
related issue: https://github.com/kyma-project/lifecycle-manager/issues/868

- the test of normal state propagation during module delete has been covered [here](https://github.com/kyma-project/lifecycle-manager/pull/956/files#diff-ea86e84ebe177150874aafaa958558196cdd8bf2461a02a11b00b5b7b03ad44bR55-R75)
- the test of warning state propagation during module delete has been covered [here](https://github.com/kyma-project/lifecycle-manager/pull/956/files#diff-02f29766ed71850fe0275f8cea90bd331073eba09d1770d43b9d2b0472c1b872R102-R131)
- refactored `helper_tests.go`, relocated more shared utility functions into `testutils`

### CODE REVIEW

File: pkg/testutils/kyma.go [hunk 1]
```diff
@@ -117,10 +198,11 @@ func IsKymaInState(ctx context.Context, name, namespace string, clnt client.Clie
 		string(state))
 }
 
-func ExpectKymaManagerField(
-	ctx context.Context, controlPlaneClient client.Client, kymaName string, managerName string,
+func ContainsExpectKymaManagerField(
```
Comment (Tomasz-Smelcerz-SAP) [inline pkg/testutils/kyma.go:201]: Consider renaming it to `ContainsKymaManagerField` or `StatusContainsManagerField`

File: pkg/testutils/kyma.go [hunk 2]
```diff
@@ -117,10 +198,11 @@ func IsKymaInState(ctx context.Context, name, namespace string, clnt client.Clie
...
+	ctx context.Context, clnt client.Client,
+	kymaName, kymaNamespace, managerName string,
 ) (bool, error) {
-	createdKyma, err := GetKyma(ctx, controlPlaneClient, kymaName, "")
+	createdKyma, err := GetKyma(ctx, clnt, kymaName, kymaNamespace)
```
Comment (Tomasz-Smelcerz-SAP) [inline pkg/testutils/kyma.go:205]: How about renaming this to just `kyma`?

File: internal/controller/kyma_controller/kyma_test.go [hunk 3]
```diff
@@ -79,12 +90,23 @@ var _ = Describe("Kyma enable one Module", Ordered, func() {
...
-			Should(Equal(string(v1beta2.StateProcessing)))
+		Eventually(IsKymaInState, Timeout, Interval).
+			WithContext(ctx).
+			WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateProcessing).
+			Should(Succeed())
 
 		By("having created new conditions in its status")
-		Eventually(GetKymaConditions, Timeout, Interval).WithArguments(kyma.GetName()).ShouldNot(BeEmpty())
+		Eventually(func() error {
+			createdKyma, err := GetKyma(ctx, controlPlaneClient, kyma.GetName(), kyma.GetNamespace())
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/controller/kyma_controller/kyma_test.go:100]: I think this code is less readable than the original one, which was a one-liner. Generally I think we should keep the test code as short as possible and possibly reflecting the "given/when/then" BDD style. IMO this code, although simple, is long enough to deserve it's own function.

File: pkg/testutils/kyma.go (hunk 1 above)
Comment (ruanxin) [reply pkg/testutils/kyma.go:201]: changed to ContainsKymaManagerField

File: internal/controller/kyma_controller/kyma_test.go (hunk 3 above)
Comment (ruanxin) [reply internal/controller/kyma_controller/kyma_test.go]: the problem of [original one](https://github.com/kyma-project/lifecycle-manager/pull/956/files#diff-886762db4cf11d4f54f47a446185f94f18c2777d10def986b78b5509c0707982L101)is it expected return empty condition when get kyma get error, which is not solid enough, but I agree it deserve a function.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**